### ISSUE TITLE
Decay at each step/iteration and at each epoch

### ISSUE THREAD

**Issue by Praful932** [ageron/handson-ml2#287]
According to Chapter 11 Training DNN, we can use the decay parameter of SGD to decay the learning rate. According to the Power Scheduling formula  `lr(t) = lr0/(1 + t/s)^c` the function depends on the step/iteration number, does that mean that if there are 2000 steps in an epoch and there are 10 epochs then the lr goes down from t=1 to t=20000 with a constant decay as a hyperparameter?

**Comment by ibeauregard** [comment]
Actually, `lr` goes down from `lr(0) (= lr0)` to `lr(20000) (= lr0/(1 + 20000/s)^c)`. Moreover, power scheduling does _not_ have a constant decay rate. Between `t = 0` and `t = s`, lr goes from `lr0` to `lr0/2^c`. That's a ratio of 1/2^c. Then, between `t = s` and `t = 2s`, lr goes from `lr0/2^c` to `lr0/3^c`, for a ratio of (2/3)^c.

Note that contrary to power scheduling, the decay rate is constant with exponential scheduling (see p. 360).

**Comment by Praful932** [comment]
> Actually, `lr` goes down from `lr(0) (= lr0)` to `lr(20000) (= lr0/(1 + 20000/s)^c)`. Moreover, power scheduling does _not_ have a constant decay rate. Between `t = 0` and `t = s`, lr goes from `lr0` to `lr0/2^c`. That's a ratio of 1/2^c. Then, between `t = s` and `t = 2s`, lr goes from `lr0/2^c` to `lr0/3^c`, for a ratio of (2/3)^c.
> 
> Note that contrary to power scheduling, the decay rate is constant with exponential scheduling (see p. 360).

Oh thanks, that helps. So in short if `s=total number of steps` at the end of fitting, the learning rate will be halved

**Comment by ibeauregard** [comment]
If `c = 1`, yes!
